tests/posterior_nn_test.py

```diff
@@ -168,13 +181,15 @@ def test_batched_mcmc_sample_log_prob_with_different_x(
 
         x_o = torch.stack([0.5 * ones(num_dim), -0.5 * ones(num_dim)], dim=0)
         # test with multiple chains to test whether correct chains are concatenated.
-        samples = posterior.sample_batched((1000,), x_o, num_chains=2, warmup_steps=500)
+        samples = posterior.sample_batched(
+            (1000,), x_o, num_chains=num_chains, warmup_steps=500
```

Comment:
for completeness here, I would suggest to use `sample_shape` instead of 1000 here. Then, after sampling, you can flatten the sample dimension and use the same test (i.e. make sure the samples mean match for each x separately)

Reply:
I think we need the `(1000,)` to have enough samples in order to get a good mean-comparison. So I suggest to just set `sample_shape=(1000,)` in line 171.

Reply:
Okay. But now the same test is repeated several times, which is annoying since this is one of the slower tests. Can we add to the `if` statement to ensure that this is not repeated for each `sample_shape` (which is overriden for this check anyway)?

Reply:
good point. A bit hacky, but we could add `and sample_shape == (5,)` to enforce that it is executed only for one of the sample shape cases?